src/Spryker/Zed/AppWebhook/Business/WebhookHandler/WebhookHandler.php

```diff
@@ -9,30 +9,79 @@
 
 use Generated\Shared\Transfer\WebhookRequestTransfer;
 use Generated\Shared\Transfer\WebhookResponseTransfer;
+use Spryker\Shared\Log\LoggerTrait;
+use Spryker\Zed\AppWebhook\AppWebhookConfig;
+use Spryker\Zed\AppWebhook\Business\Exception\AllowedNumberOfRetriesExceededException;
+use Spryker\Zed\AppWebhook\Business\Identifier\IdentifierBuilderInterface;
+use Spryker\Zed\AppWebhook\Business\MessageBuilder\MessageBuilder;
+use Spryker\Zed\AppWebhook\Persistence\AppWebhookEntityManagerInterface;
+use Throwable;
 
 class WebhookHandler
 {
+    use LoggerTrait;
+
     /**
      * @param array<\Spryker\Zed\AppWebhook\Dependency\Plugin\WebhookHandlerPluginInterface> $webhookHandlerPlugins
      */
-    public function __construct(protected array $webhookHandlerPlugins)
-    {
+    public function __construct(
+        protected array $webhookHandlerPlugins,
+        protected AppWebhookConfig $appWebhookConfig,
+        protected AppWebhookEntityManagerInterface $appWebhookEntityManager,
+        protected IdentifierBuilderInterface $identifierBuilder
+    ) {
     }
 
     public function handleWebhook(WebhookRequestTransfer $webhookRequestTransfer, WebhookResponseTransfer $webhookResponseTransfer): WebhookResponseTransfer
     {
-        foreach ($this->webhookHandlerPlugins as $webhookHandlerPlugin) {
-            if (!$webhookHandlerPlugin->canHandle($webhookRequestTransfer)) {
-                continue;
-            }
+        // get the default identifier which will be only used when the webhook is not a retry
+        $identifier = $this->identifierBuilder->getIdentifier();
+
+        // Only persist the webhook if it's not a retry.
+        if ($webhookRequestTransfer->getIsRetry() !== true) {
```

Comment:
If the identifier is a marker for an already persisted webhook, why do we need `isRetry`?

Reply:
To know it it is a new one or if it was already tried to be executed. The isRetry is set when getting all related webhooks and to run them again. Every WebhookRequest will get an identifier and we need to be able to know if it is a new one or one that we retry.